File: app/src/main/java/com/orego/corporation/orego/fragments/otherActivities/face3dActivity/model3D/view/ModelSurfaceView.java
package com.orego.corporation.orego.fragments.otherActivities.face3dActivity.model3D.view;

import android.annotation.SuppressLint;
import android.opengl.GLSurfaceView;
import android.view.MotionEvent;

import com.orego.corporation.orego.fragments.otherActivities.face3dActivity.model3D.controller.TouchController;
import com.orego.corporation.orego.fragments.otherActivities.face3dActivity.model3D.modelRender.ModelRender;
import com.orego.corporation.orego.fragments.otherActivities.face3dActivity.model3D.portrait.headComposition.HeadComposition;

@SuppressLint("ViewConstructor")
public final class ModelSurfaceView extends GLSurfaceView {

    private ModelRender mRenderer;
    private ModelActivity parent;
    private TouchController touchHandler;

    public ModelSurfaceView(final ModelActivity parent, HeadComposition headComposition) {
        super(parent);
        this.parent = parent;
        setEGLContextClientVersion(3);
        mRenderer = new ModelRender(this, headComposition);
        setRenderer(mRenderer);
        touchHandler = new TouchController(this, mRenderer);

    }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public boolean onTouchEvent(MotionEvent event) {
        return touchHandler.onTouchEvent(event);
    }


    public ModelActivity getModelActivity() {
        return parent;
    }

    public ModelRender getmRenderer() {
        return mRenderer;
    }

}

File: app/src/main/java/com/orego/corporation/orego/views/modelFragment/modelView/ModelSurfaceView.kt
package com.orego.corporation.orego.views.modelFragment.modelView

import android.annotation.SuppressLint
import android.opengl.GLSurfaceView
import android.view.MotionEvent
import com.orego.corporation.orego.presenters.MainActivity
import com.orego.corporation.orego.models.controllers.TouchController
import com.orego.corporation.orego.presenters.modelRender.ModelRender
import com.orego.corporation.orego.models.portrait.headComposition.HeadComposition

@SuppressLint("ViewConstructor")
class ModelSurfaceView (mainActivity: MainActivity, headComposition: HeadComposition): GLSurfaceView(mainActivity) {
    private var mRenderer: ModelRender
    private val parent = mainActivity
    private var touchHandler: TouchController

    init{
        setEGLContextClientVersion(3)
        mRenderer = ModelRender(this, headComposition)
        setRenderer(mRenderer)
        touchHandler = TouchController(this, mRenderer)

    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(event: MotionEvent): Boolean {
        return touchHandler.onTouchEvent(event)
    }


    fun getMainActivity(): MainActivity {
        return parent
    }

}